android/java/org/chromium/chrome/browser/shields/BraveShieldsHandler.java

```diff
@@ -182,27 +184,30 @@ public BraveShieldsHandler(Context context) {
 
     public void loadDisconnectEntityList(Context context) {
         if (context == null) return;
-        try {
-            String jsonString = loadDisconnectEntityJSONFromAsset(context);
-            if (jsonString == null) return;
-            JSONObject obj = new JSONObject(jsonString);
-            JSONObject entities = obj.getJSONObject("entities");
-            Iterator<String> keysItr = entities.keys();
-            while (keysItr.hasNext()) {
-                String key = keysItr.next();
-                Object value = entities.get(key);
-                JSONArray jsonProperties = ((JSONObject) value).getJSONArray("properties");
-                JSONArray jsonResources = ((JSONObject) value).getJSONArray("resources");
-
-                for (int i = 0; i < jsonResources.length(); i++) {
-                    mResourceToCompanyNameList.add(new Pair(jsonResources.getString(i), key));
+        ExecutorService executor = Executors.newSingleThreadExecutor();
```

Comment:
let's better use chromium's `PostTask` similar to this https://github.com/brave/brave-core/blob/master/android/java/org/chromium/chrome/browser/app/BraveActivity.java#L783. Google made a special class for that purpose that is optimised to be used inside the browser. We should use it instead of doing our own implementation.